core/cloudflow-akka/src/main/scala/cloudflow/akkastream/AkkaStreamletContextImpl.scala

```diff
@@ -362,18 +381,49 @@ final class AkkaStreamletContextImpl(
     // we created this file when the pod started running (see AkkaStreamlet#run)
     Files.deleteIfExists(Paths.get(s"/tmp/$streamletRef.txt"))
 
-    killSwitch.shutdown()
     import system.dispatcher
+    val controls = KafkaControls.get
```

Comment:
`stop` is currently only called on AkkaStreamlet catching exception in run. This logic should also happen if it completes without error.